spyder/app/utils.py

```diff
@@ -325,3 +334,59 @@ def create_window(WindowClass, app, splash, options, args):
     if not running_under_pytest():
         app.exec_()
     return main
+
+
+class SpyderInstallerError(object):
```

Comment:
Let's move this and the other classes you added here to `spyder/utils/installers.py`. That's because they are not used in `app/mainwindow.py` but all over the place instead.